Source/WTF/wtf/glib/GSpanExtras.h

```diff
@@ -21,9 +21,28 @@
 
 #include <wtf/StdLibExtras.h>
 #include <wtf/glib/GRefPtr.h>
+#include <wtf/glib/GUniquePtr.h>
 
 namespace WTF {
 
+template <typename T, size_t Extent = std::dynamic_extent>
+class GOwningSpan : public std::span<T> {
+    WTF_MAKE_NONCOPYABLE(GOwningSpan);
```

Comment:
I know this is non-copyable but presumably this is still movable? Wouldn't the destructor of the moved-out `GOwningSpan` still free the data pointer as currently implemented? If so, couldn't this still lead to double-frees?

Reply:
Actually I think it's non-movable because (a) copy constructor and copy assignment operator are deleted, and (b) move constructor and move assignment operator are not defined.

But attempting to make it noncopyable is entirely futile, because std::span is copyable. I'm disappointed that C++ allows this. I'm not even sure what will happen when the base class copy constructor or copy assignment operator gets called; I guess the base class gets copied and the subclass just doesn't?

Reply:
@cdumez Thanks for the heads up, the changes you want to do to `MallocPtr` makes it look like something we could use in this case. I'll set this PR as draft and rework it later on top of `MallocPtr`/`MallocSpan` when it's ready.

Reply:
Ok. I'll prioritize getting MallocSpan landed so as to not block people for too long.

Reply:
https://github.com/WebKit/WebKit/pull/35950